OADKit/OADFirmwareImage.swift

```diff
@@ -216,38 +222,39 @@ class FirmwareImage : NSObject {
         
         var page = startPage
         var pageEnd = (Int)(len / (0x1000 / 4))
-        let osetEnd = ((len - (pageEnd * (0x1000 / 4))) * 4)
+        let osetEnd = (len - (pageEnd * (0x1000 / 4))) * 4
         
         pageEnd += startPage
         
-        while (true) {
+        while true {
             var oset = 0
             while oset < 0x1000 {
-                if ((page == startPage) && (oset == 0x00)) {
+
+                if (page == startPage) && (oset == 0x00) {
+
                     //Skip the CRC and shadow.
                     //Note: this increments by 3 because oset is incremented by 1 in each pass
                     //through the loop
                     oset += 3
-                }
-                
-                else if ((page == pageEnd) && (oset == osetEnd)) {
+
+                } else if (page == pageEnd) && (oset == osetEnd) {
+
                     crc = crc16(startCrc: crc, startVal: 0x00)
                     crc = crc16(startCrc: crc, startVal: 0x00)
-        
-                    oset += 1
                     return crc
-                }
-                    
-                else {
-                    data.withUnsafeBytes {(bytes: UnsafePointer<UInt8>) in
+
+                } else {
+
+                    data.withUnsafeBytes { (bytes: UnsafePointer<UInt8>) in
                         crc = crc16(startCrc: crc, startVal: Int(bytes[Int(addr + oset)]))
                     }
                 }
+
+                oset += 1
             }
         
             page += 1
             addr = page * 0x1000
-            oset += 1
```

Comment:
Im not sure about moving this inside the else as in the original version (pre swift 3, with for loop) oset would be incremented by 1 each loop, regardless of the start page's increment by 3.